### PR TITLE
dirty lines instead of printing spaces

### PR DESCRIPTION
**Pull request by lastquestion**
This is way cleaner then the printing white space approach.

1. Use `erase` instead of `clear`. `erase` does not call `refresh`, whereas `clear` does. We need to draw before refreshing anyway, so we save on a `refresh` call which is (a) cleaner and (b) potentially flicker free on slow SSH connections, for ex.

2. We use delch to delete and 'reset' a line. It removes the character and the associated attribute at some index. I've also mocked this to mean exactly that in the test suite, so we don't need to have spaces anymore in the tests.

3. Rename `dirtyLines` to `dirtyAll` to match what it does: it dirties evrything and forces chrome to redraw.

4. Pep8 pass.

### CODE REVIEW

**Comment by lastquestion** [comment]
I plan to open a follow on PR to use `pad` so we can actually paint the entire scroll buffer and cache it; `curses` takes care of painting onto the visible screen via `refresh`. It will greatly simplify the painting code and also be faster - we will never write a line twice until it changes highlight/hover/select state. I need to do some investigation on size limits though - like I'll try it with 10/50/100k lines and so, make sure it is performant. If so, it'll clean up a lot of code nicely.

File: src/format.py [hunk 1]
```diff
@@ -45,13 +45,16 @@ class LineMatch(object):
...
+
         self.formattedLine = formattedLine
         self.index = index
 
         (file, num, matches) = result
 
         self.originalFile = file
-        self.file = parse.prependDir(file, withFileInspection=validateFileExists)
+        self.file = parse.prependDir(file,
+                                     withFileInspection=validateFileExists)
```
Comment (pcottle) [inline src/format.py]: why this change? was it over 80char?

File: src/screenControl.py [hunk 2]
```diff
@@ -213,19 +213,20 @@ def __init__(self, flags, stdscr, lineObjs, cursesAPI):
         (self.oldmaxy, self.oldmaxx) = self.getScreenDimensions()
         self.mode = SELECT_MODE
 
-        self.simpleLines = []
```
Comment (pcottle) [inline src/screenControl.py]: do we not use this for anything? :O

File: src/screenControl.py [hunk 3]
```diff
@@ -490,27 +486,34 @@ def onEnter(self):
...
-        if self.linesDirty:
+        if self.dirty:
             self.printAll()
-        for index in self.dirtyIndexes:
-            self.lineMatches[index].output(self.colorPrinter)
-        if self.helperChrome.getIsSidebarMode():
-            # need to output since lines can override
-            # the sidebar stuff
-            self.printChrome()
+        else:
```
Comment (pcottle) [inline src/screenControl.py]: nit: the above could just be an early return so we dont indent too much

File: src/screenControl.py [hunk 4]
```diff
@@ -490,27 +486,34 @@ def onEnter(self):
...
+                y = miny + index + self.getScrollOffset()
+                if y >= miny or y < maxy:
+                    self.clearLine(y)
+                    self.lineObjs[index].output(self.colorPrinter)
+
+    def clearLine(self, y):
+        '''Clear a line of content, excluding the chrome'''
+        (minx, miny, maxx, maxy) = self.getChromeBoundaries()
+        for x in range(minx, maxx):
+            self.stdscr.delch(y, x)
```
Comment (pcottle) [inline src/screenControl.py]: ahhhh cool!!!!

**Comment by pcottle** [comment]
:tada: :tada: :tada: 
Woohoo! This is way better -- deleting that hacky whitespace at the end of the expected test output feels quite nice :D

I didn't know about delchar to be honest, clearly I should have read through more of the curses documentation. This is pretty sweet though, I like how we re-use the chrome boundaries to understand which lines to clear

Lemme merge this and followup with some nits

File: src/__tests__/expected/selectDownSelectInverse.txt [hunk 5]
```diff
@@ -1,6 +1,6 @@
- README.md                      |  8 ++++-     
+ README.md                      |  8 ++++-
  fpp                            |  6 ++--
- src/__tests__/__init__.py      |  0
```
Comment (pcottle) [inline src/__tests__/expected/selectDownSelectInverse.txt]: :heart_eyes:

File: src/format.py (hunk 1 above)
Comment (lastquestion) [inline src/format.py]: My version of pep8 complained and broke the line with `/`, which I didn't like so I used a new line instead. What version do you have installed?

File: src/screenControl.py (hunk 2 above)
Comment (lastquestion) [inline src/screenControl.py]: haha nah. I was surprised and double-grepped to be sure

File: src/screenControl.py (hunk 3 above)
Comment (lastquestion) [inline src/screenControl.py]: Yah. I'll change it around next PR

File: src/__tests__/expected/selectDownSelectInverse.txt (hunk 5 above)
Comment (lastquestion) [inline src/__tests__/expected/selectDownSelectInverse.txt]: The best. :tada:
